packages/eslint-plugin/typings/eslint-rules.d.ts

```diff
@@ -817,17 +817,90 @@ declare module 'eslint/lib/rules/no-invalid-this' {
+
+      /**
+       * The second parameter of `node` (with a type of `TSESTree.Node`) is
+       * [intentionally omitted in order to prevent TypeScript
+       * errors](https://github.com/typescript-eslint/typescript-eslint/issues/6993).
+       *
+       * If you want to use this method, you can either ignore the `node`
+       * parameter or specify it with a type assertion, depending on what you
+       * want to do.
+       */
+      onCodePathStart?: (
+        codePath: TSESLint.CodePath,
+        // node: TSESTree.Node,
+      ) => void;
+
+      /**
+       * The second parameter of `node` (with a type of `TSESTree.Node`) is
+       * [intentionally omitted in order to prevent TypeScript
+       * errors](https://github.com/typescript-eslint/typescript-eslint/issues/6993).
+       *
+       * If you want to use this method, you can either ignore the `node`
+       * parameter or specify it with a type assertion, depending on what you
+       * want to do.
+       */
+      onCodePathEnd?: (
+        codePath: TSESLint.CodePath,
+        // node: TSESTree.Node,
+      ) => void;
+
+      /**
+       * The second parameter of `node` (with a type of `TSESTree.Node`) is
+       * [intentionally omitted in order to prevent TypeScript
+       * errors](https://github.com/typescript-eslint/typescript-eslint/issues/6993).
+       *
+       * If you want to use this method, you can either ignore the `node`
+       * parameter or specify it with a type assertion, depending on what you
+       * want to do.
+       */
+      onCodePathSegmentStart?: (
+        segment: TSESLint.CodePathSegment,
+        // node: TSESTree.Node,
+      ) => void;
+
+      /**
+       * The second parameter of `node` (with a type of `TSESTree.Node`) is
+       * [intentionally omitted in order to prevent TypeScript
+       * errors](https://github.com/typescript-eslint/typescript-eslint/issues/6993).
+       *
+       * If you want to use this method, you can either ignore the `node`
+       * parameter or specify it with a type assertion, depending on what you
+       * want to do.
+       */
+      onCodePathSegmentEnd?: (
+        segment: TSESLint.CodePathSegment,
+        // node: TSESTree.Node,
+      ) => void;
+
+      /**
+       * The second parameter of `toSegment` (with a type of
+       * `TSESLint.CodePathSegment`) and the third parameter of `node` (with a
+       * type of `TSESTree.Node`) is [intentionally omitted in order to prevent
+       * TypeScript
+       * errors](https://github.com/typescript-eslint/typescript-eslint/issues/6993).
+       *
+       * If you want to use this method, you can either ignore those parameters
+       * or specify them with a type assertion, depending on what you want to
+       * do.
+       */
+      onCodePathSegmentLoop?: (
+        fromSegment: TSESLint.CodePathSegment,
+        // toSegment: TSESLint.CodePathSegment,
+        // node: TSESTree.Node,
+      ) => void;
```

Comment:
we don't need to add these here.
our extension rule doesn't care about the code path selectors so no need to complicate things

Reply:
disagree - I think it is good for DX, as it adds auto-complete to VSCode and is a valuable indication that these methods actually exist, as well as provide good mouse-over documentation on exactly what to do if you need to use them.

Reply:
This particular set of types is ***ONLY*** used internally within our project and it exists to let us expose the selectors that this particular rule has so that we can call them within our extension version of the rule.

But because our extension version does not touch the code path analysis - these aren't used. So uncommenting them does not change or fix anything.

Reply:
ok, i will delete them then.
can you tell me how to make autocomplete work though for plugin authors?

Reply:
we can't do autocomplete as that would require adding the signatures to the selectors object.
the types you've added are in the realm of "good enough" to allow people to define their rule in the very rare case that they want the code path analysis.

Reply:
ok